Angular Testing Masterclass/README.md

```diff
@@ -757,8 +757,163 @@
+- Cypress was originally designed to run end-to-end (E2E) tests on anything that runs in a browser. A typical E2E test visits the application in a browser and performs actions via the UI just like a real user would.
+> It's about testing how it all fits together. This mean that we are actually testing our complete Frontend application as it's going to be deployed in production.
+> - NO mocking of services involved in this type of test.
+- Installing Cypress
+    - `npm install cypress --save-dev` This will install Cypress locally as a dev dependency for your project.
+- Adding npm Scripts
+      ```       
+        {
+            "scripts": {
+                "cypress:open": "cypress open"
+                "cypress:run":  "cypress run"
+            }
+        }
+      ``` 
+    > cypress:open is using to run cypress UI 
+ - cypress.json
+   - This file contains the baseUrl, This url must be the same url where the project is deployed locally.
+     ```
+       {
+           "baseUrl": "http://localhost:4200"
+       }      
+     ```      
+ - Add a test file
+   - The test file should be create inside integration directory `cypress/integration/home.test.js`   
+   - The json files with mock data should be create inside fixture directory `cypress/fixtures/courses.json` to simulate the HTTP response.
+    > fixture: Load a fixed set of data located in a file. 
+- Basic suite test
+   -  If the application is running locally and the cypress.json baseUrl is pointing to localhost and the same port, then we can use the `cy.visit('/')` global variable to visit the route page of our application.  
+   -  cy.contains: Get the DOM element containing the text.`cy.contains("All Courses")` DOM elements can contain more than the desired text and still match.   
+   ```ruby
+     describe('Component Page', () => {            
+
+            it('should display Title', () => {
+
+                cy.visit('/');
+                cy.contains("All Courses");               
+
+            }); 
+        }); 
+
+   ``` 
+
+
+
+
+
+
+- Home component E2E test
+  - .fixture( ): Load data from file.
+  - .as( ): Assign an alias for later use.
+    - Reference the alias later within a cy.get( ) or cy.wait( ) command with an @ prefix. 
+  - .server( ): Start a server to begin routing responses to cy.route( ) and to change the behavior of network requests.
+  > cy.server( ) and cy.route( ) are deprecated in Cypress 6.0.0. In a future release, support will be removed. 
+  > -Consider using cy.intercept( ) instead. 
+  - .wait( ): Wait for a number of milliseconds or wait for an aliased resource to resolve before moving on to the next command.
+  - .get( ): Get one or more DOM elements by selector or alias.
+  - .last( ): Get the last DOM element within a set of DOM elements.
+  - .its( ): Get a property's value on the previously yielded subject.
+  - .should( ): Create an assertion. 
+  ```ruby
```

Comment:
Change the MD template syntax to `js` instead of `ruby`

Reply:
Hi, @alexeikmun
The MD template syntax was changed